src/main/java/org/testng/TestRunner.java

```diff
@@ -792,10 +806,10 @@ private void privateRun(XmlTest xmlTest) {
 
     // Check if an interceptor had altered the effective test method count. If yes, then we need to
     // update our configurationGroupMethod object with that information.
-    if (resultArray.length != m_groupMethods.getAllTestMethods().length) {
+    if (resultArray.length != testMethodsContainer.getItems().length) {
       m_groupMethods =
           new ConfigurationGroupMethods(
-              resultArray,
+              new TestMethodContainer(resultArray),
```

Comment:
I see a potential issue because our previous assumption was: TestRunner and ConfigurationGroupMethods share the same method cache.
But here, we see it is not the case because we are creating a new ConfigurationGroupMethods with another cache.

Reply:
In fact, it only shows that m_groupMethods was not a good object for the cache.

So now, `TestMethodContainer` is just another way to restore `m_allTestMethods` and a wrapper object is no more a big help.
We can keep it or not, I let you the choice.

Reply:
@juherr - True that. But I dont get as to what the potential issue now you are talking about because we now have gotten rid of the side effect of relying on `ConfigurationGroupMethods`. So I didnt quite understand what should I be fixing here ?

Reply:
There is nothing to change here.
The potential issue was if the expected method list had to be the same in both cases. It looks it is not the case.

As it is not the case, `TestMethodContainer` is no more useful.

Reply:
Ok. So I guess this was an observation behind the ask for creating a separate cache/container for holding test methods so that we rule out all side effects and we stick to single responsibility principle. Point taken. Resolving this conversation

Reply:
I asked for a distinct object cache because my understanding at this moment was that the cache was a shared thing.
Your previous commit highlighted that it was not the case and my ask was less pertinent.

My current conclusion is: there are 2 different methods cache at 2 different places and we cannot merge their usage into a single place.

Reply:
@juherr - yes that is correct. I guess I may have opened up the pandora's box for all the wrong reasons :) But that's fine. We can iterate on this to streamline and make things better. In the meantime, please do take another look at the latest changes that I now added as an additional commit.

Reply:
There are many pandora boxes everywhere. :D And my objective is always to limit the little workarounds which are making more complexity on the whole picture! ;)

Reply:
True that and completely agree with you on that. Apart from the formatting changes, are there any other things that need to be taken care of ?

Reply:
> are there any other things that need to be taken care of?

No. Feel free to merge :)

Reply:
I have merged the changes. Thanks for helping me out with reviewing this PR @juherr